control-plane/p4RuntimeSerializer.cpp

```diff
@@ -604,6 +606,38 @@ getActionRefs(const IR::P4Table* table, ReferenceMap* refMap) {
     return actions;
 }
 
+static cstring
+getMatchTypeName(const IR::PathExpression* matchPathExpr, const ReferenceMap* refMap) {
+    CHECK_NULL(matchPathExpr);
+    auto matchTypeDecl = refMap->getDeclaration(matchPathExpr->path, true)
+        ->to<IR::Declaration_ID>();
+    BUG_CHECK(matchTypeDecl != nullptr, "No declaration for match type '%1%'", matchPathExpr);
+    return matchTypeDecl->name.name;
+}
+
+/// maps the match type name to the corresponding P4Info MatchType enum
+/// member. If the match type should not be exposed to the control plane and
+/// should be ignored, boost::none is returned. If the match type does not
+/// correspond to any standard match type known to P4Info, default enum member
+/// UNSPECIFIED is returned.
+static boost::optional<MatchField::MatchType>
+getMatchType(cstring matchTypeName) {
+    if (matchTypeName == P4CoreLibrary::instance.exactMatch.name) {
+        return MatchField::MatchTypes::EXACT;
+    } else if (matchTypeName == P4CoreLibrary::instance.lpmMatch.name) {
+        return MatchField::MatchTypes::LPM;
+    } else if (matchTypeName == P4CoreLibrary::instance.ternaryMatch.name) {
+        return MatchField::MatchTypes::TERNARY;
+    } else if (matchTypeName == P4V1::V1Model::instance.rangeMatchType.name) {
+        return MatchField::MatchTypes::RANGE;
+    } else if (matchTypeName == P4V1::V1Model::instance.selectorMatchType.name) {
+        // Nothing to do here, we cannot even perform some sanity-checking.
+        return boost::none;
+    } else {
+        return MatchField::MatchTypes::UNSPECIFIED;
```

Comment:
Do you really need UNSPECIFIED? The spec says: "If the @match annotation is not specified on a struct field, by default it is assumed to be @match(exact)." Shouldn't then just return EXACT?